# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CSSEditUtils.cpp
# Commit: 3efbfdf889b4
# Full Hash: 3efbfdf889b4bcda432a73871eb329523f78357b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 09:46:07
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 7-2: Make `CSSEditUtils::RemoveCSSInlineStyleWithTransaction` stop touching `Selection` directly r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152975
# ==============================================================================

diff -r f9d7d41f5043 -r 3efbfdf889b4 editor/libeditor/CSSEditUtils.cpp
--- a/editor/libeditor/CSSEditUtils.cpp	Thu Aug 04 04:19:46 2022 +0000
+++ b/editor/libeditor/CSSEditUtils.cpp	Thu Aug 04 04:28:13 2022 +0000
@@ -571,7 +571,8 @@
 
 // remove the CSS style "aProperty : aPropertyValue" and possibly remove the
 // whole node if it is a span and if its only attribute is _moz_dirty
-nsresult CSSEditUtils::RemoveCSSInlineStyleWithTransaction(
+Result<EditorDOMPoint, nsresult>
+CSSEditUtils::RemoveCSSInlineStyleWithTransaction(
     nsStyledElement& aStyledElement, nsAtom* aProperty,
     const nsAString& aPropertyValue) {
   // remove the property from the style attribute
@@ -579,30 +580,20 @@
                                                  aPropertyValue);
   if (NS_FAILED(rv)) {
     NS_WARNING("CSSEditUtils::RemoveCSSPropertyWithTransaction() failed");
-    return rv;
+    return Err(rv);
   }
 
   if (!aStyledElement.IsHTMLElement(nsGkAtoms::span) ||
       HTMLEditor::HasAttributes(&aStyledElement)) {
-    return NS_OK;
+    return EditorDOMPoint();
   }
 
   OwningNonNull<HTMLEditor> htmlEditor(*mHTMLEditor);
-  const Result<EditorDOMPoint, nsresult> unwrapStyledElementResult =
+  Result<EditorDOMPoint, nsresult> unwrapStyledElementResult =
       htmlEditor->RemoveContainerWithTransaction(aStyledElement);
-  if (MOZ_UNLIKELY(unwrapStyledElementResult.isErr())) {
-    NS_WARNING("HTMLEditor::RemoveContainerWithTransaction() failed");
-    return unwrapStyledElementResult.inspectErr();
-  }
-  const EditorDOMPoint& pointToPutCaret = unwrapStyledElementResult.inspect();
-  if (!htmlEditor->AllowsTransactionsToChangeSelection() ||
-      !pointToPutCaret.IsSet()) {
-    return NS_OK;
-  }
-  rv = htmlEditor->CollapseSelectionTo(pointToPutCaret);
-  NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
-                       "EditorBase::CollapseSelectionTo() failed");
-  return rv;
+  NS_WARNING_ASSERTION(unwrapStyledElementResult.isOk(),
+                       "HTMLEditor::RemoveContainerWithTransaction() failed");
+  return unwrapStyledElementResult;
 }
 
 // Answers true if the property can be removed by setting a "none" CSS value